icaruscode/CRT/CRTCalibrationAnalysis_module.cc

```diff
@@ -0,0 +1,338 @@
+ 
+  CRTCalibrationAnalysis::CRTCalibrationAnalysis(Parameters const& config)
+    : EDAnalyzer(config)
+    , fCRTDAQProducerLabel(config().CRTDAQLabel())
+    , fCrtutils(new CRTCommonUtils())
+  {
+    // Get a pointer to the geometry service provider.
+    fGeometryService = lar::providerFrom<geo::Geometry>();
+    // The same for detector TDC clock services.
+    // Access to detector properties.
+    auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataForJob();
+    fTriggerOffset = trigger_offset(clockData);
+
+	for(int i=1; i<94; i++){
+
+		macToHistos[i] = new vector<TH1F*>();
+
+		for(int ch=0; ch<32; ch++){
+
+			string hname = "hadc_"+to_string(i)+"_"+to_string(ch);
+			string htitle = "raw charge: mac5 "+to_string(i)+", ch. "+to_string(ch);
+			macToHistos[i]->push_back(tfs->make<TH1F>(hname.c_str(),htitle.c_str(),4100,0,4100));
+//			macToHistos[i]->at(ch)->Write();
+		}
+	}
+  }
+
+  void CRTCalibrationAnalysis::FillFebMap() { 
+    if(!this->fFebMap.empty())
+        return;
+    std::string fullFileName;
+    cet::search_path searchPath("FW_SEARCH_PATH");
+    searchPath.find_file("feb_map.txt",fullFileName);
+    std::ifstream fin;
+    fin.open(fullFileName,std::ios::in);
+    if(fin.good()) std::cout << "opened file 'feb_map.txt' for reading..." << std::endl;
+    else //std::cout << "could not open file 'feb_map.txt' for reading!" << std::endl;
+        throw cet::exception("CRTDataAnalysis::FillFebMap") << "Unable to find/open file 'feb_map.txt'" << std::endl;
+    std::vector<std::string> row;
+    std::string line, word;
+    while(getline(fin,line)) {
+        row.clear();
+        std::stringstream s(line);
+        int mod;
+        while (std::getline(s, word, ',')) {
+            row.push_back(word);
+        }
+        mod = std::stoi(row[0]);
+        (this->fFebMap)[mod].push_back(std::make_pair(std::stoi(row[1]),std::stoi(row[2])));
+        if(row.size()>3)
+            (this->fFebMap)[mod].push_back(std::make_pair(std::stoi(row[3]),std::stoi(row[4])));
+    }
+    std::cout << "filled febMap with " << (this->fFebMap).size() << " entries" << std::endl;
+    fin.close();
+  }
+
+  
+  //-----------------------------------------------------------------------
+  void CRTCalibrationAnalysis::beginJob()
+  {
+    std::cout << " starting analysis job" << std::endl;
+
+}
+   
+  void CRTCalibrationAnalysis::beginRun(const art::Run& /*run*/)
+  {
+  }
+
+  //-----------------------------------------------------------------------
+  void CRTCalibrationAnalysis::analyze(const art::Event& event) 
+  {
+    MF_LOG_DEBUG("CRT") << "beginning analyis" << '\n';
+
+    // Start by fetching some basic event information for our n-tuple.
+    fEvent  = event.id().event(); 
+    fRun    = event.run();
+    fSubRun = event.subRun();
+
+	std::cout << "event: " << fEvent << ", run " << fRun << ", subrun " << fSubRun << " found." << std::endl;
```

Comment:
I would recommend to remove this std::cout line, as it will create lot of verbose and slow down the process.